modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java

```diff
@@ -394,24 +393,66 @@ public String modelFileFolder() {
         return outputFolder + File.separator + sourceFolder + File.separator + modelPackage().replace('.', File.separatorChar);
     }
 
+    private enum KotlinType {
```

Comment:
KotlinType is the way generator advises templates to implement model as. It can advise to implement it as `interface` (because model has implementers), or as `open class` (when there is only children), or as `data class` (when there are no children).

Reply:
Maybe we should rename it to something like `ModelType` or `KotlinModelType`?

Reply:
> Maybe we should rename it to something like `ModelType` or `KotlinModelType`?
Renamed to `KotlinModelType` and updated vendor extension name as well (to `x-kotlin-model-type`)